src/Propulsion.SqlStreamStore/SqlStreamStoreSource.fs

```diff
@@ -0,0 +1,57 @@
+namespace Propulsion.SqlStreamStore
+
+open System
+open SqlStreamStore
+
+open StreamReader
+open LockedIngester
+
+type SqlStreamStoreSource =
+
+    static member Run
+            (logger : Serilog.ILogger,
+             store: IStreamStore,
+             ledger: ILedger,
+             sink : Propulsion.ProjectorPipeline<_>,
+             ?consumerGroup: string,
+             ?statsInterval: TimeSpan,
+             ?readerMaxBatchSize: int,
+             ?readerSleepInterval: TimeSpan,
+             ?boundedCapacity: int,
+             ?onCaughtUp: unit -> unit) : Async<unit> =
+
+        async {
+            let consumerGroup = defaultArg consumerGroup "default"
+            let streamId = "$all"
+
+            let boundedCapacity = defaultArg boundedCapacity 10
+
+            let logger =
+                let instanceId = System.Guid.NewGuid()
+                logger
+                    .ForContext("instanceId", string instanceId)
+                    .ForContext("consumerGroup", consumerGroup)
+
+            let ingester : Propulsion.Ingestion.Ingester<_,_> =
```

Comment:
Should the ingester be explicitly stopped in a finally clause? I didn't see that done elsewhere in Propulsion and I'm not clear on who should manage this object's lifetime.